src/reader/linearring.rs

```diff
@@ -118,3 +115,135 @@ impl<'a> LineStringTrait for &LinearRing<'a> {
         )
     }
 }
+
+impl<'a> GeometryTrait for LinearRing<'a> {
+    type T = f64;
+    type PointType<'b>
+        = Point<'a>
+    where
+        Self: 'b;
+    type LineStringType<'b>
+        = LinearRing<'a>
+    where
+        Self: 'b;
+    type PolygonType<'b>
+        = Polygon<'a>
+    where
+        Self: 'b;
+    type MultiPointType<'b>
+        = MultiPoint<'a>
+    where
+        Self: 'b;
+    type MultiLineStringType<'b>
+        = MultiLineString<'a>
+    where
+        Self: 'b;
+    type MultiPolygonType<'b>
+        = MultiPolygon<'a>
+    where
+        Self: 'b;
+    type GeometryCollectionType<'b>
+        = GeometryCollection<'a>
+    where
+        Self: 'b;
+    type RectType<'b>
+        = geo_traits::UnimplementedRect<f64>
+    where
+        Self: 'b;
+    type LineType<'b>
+        = geo_traits::UnimplementedLine<f64>
+    where
+        Self: 'b;
+    type TriangleType<'b>
+        = geo_traits::UnimplementedTriangle<f64>
+    where
+        Self: 'b;
+
+    fn dim(&self) -> geo_traits::Dimensions {
+        self.dimension().into()
+    }
+
+    fn as_type(
+        &self,
+    ) -> geo_traits::GeometryType<
+        '_,
+        Self::PointType<'a>,
+        Self::LineStringType<'a>,
+        Self::PolygonType<'a>,
+        Self::MultiPointType<'a>,
+        Self::MultiLineStringType<'a>,
+        Self::MultiPolygonType<'a>,
+        Self::GeometryCollectionType<'a>,
+        Self::RectType<'_>,
+        Self::TriangleType<'_>,
+        Self::LineType<'_>,
```

Comment:
I could use a second pair of eyes here. Some of these use the inferred lifetime and some of them use `<'a>`. I don't know which is right, or if they're both right?

There are a few places throughout the codebase that have similar situations

Reply:
I think #62 still relies on unreleased features in `wkt` that were added since 0.13 was released (we haven't released wkt 0.14 yet)

Reply:
Also, please be aware that in https://github.com/georust/wkb/pull/62 I am using the concrete types such as  `Point<'a>` instead of using GAT (`Self::PointType<'_>`). That's why #62 appears to use `'a`, but when we switch to use GAT, the situation will become very different.

Reply:
> Also, please be aware that in #62 I am using concrete types such as `Point<'a>` instead of using GAT (`Self::PointType<'_>`), that's why #62 appears to use `'a`. But when we switch to use GAT, the situation will become very different.

Is it better to use concrete types than the GATs? Should I switch this PR to use the concrete types with the named `'a` lifetime?

Reply:
Thank you!